plutus-pab/src/Cardano/Node/Types.hs

```diff
@@ -46,19 +81,113 @@ data MockServerConfig =
+        BlockOperation e          -> "Block operation " <+> pretty e
+        CreatingRandomTransaction -> "Generating a random transaction"
+        FollowerMsg m             -> pretty m
+
+instance ToObject MockServerLogMsg where
+    toObject v = \case
+        NoRandomTxGeneration      ->  mkObjectStr "Not creating random transactions" ()
+        StartingRandomTx          ->  mkObjectStr "Starting random transaction generation thread" ()
+        KeepingOldBlocks          ->  mkObjectStr "Not starting block reaper thread (old blocks will be retained in-memory forever" ()
+        RemovingOldBlocks         ->  mkObjectStr "Starting block reaper thread (old blocks will be removed)" ()
+        StartingMockServer p      ->  mkObjectStr "Starting Mock Node Server on port " (Tagged @"port" p)
+        StartingSlotCoordination  ->  mkObjectStr "" ()
+        ProcessingChainEvent e    ->  mkObjectStr "Processing chain event" (Tagged @"event" e)
+        BlockOperation e          ->  mkObjectStr "Block operation" (Tagged @"event" e)
+        CreatingRandomTransaction ->  mkObjectStr "Creating random transaction" ()
+        FollowerMsg m             -> toObject v m
+
+-- | logging type for 'Cardano.Node.Follower'
+data NodeFollowerLogMsg =
+    NewFollowerId FollowerID
+    | GetBlocksFor FollowerID
+    | LastBlock Int
+    | NewLastBlock Int
+    | GetCurrentSlot Slot
+    deriving (Show, Eq, Generic, FromJSON, ToJSON)
+
+instance Pretty NodeFollowerLogMsg where
+    pretty  = \case
+        NewFollowerId newID -> "New follower ID:" <+> pretty newID
+        GetBlocksFor i      -> "Get blocks for" <+> pretty i
+        LastBlock i         -> "Last block:" <+> pretty i
+        NewLastBlock i      -> "New last block:" <+> pretty i
+        GetCurrentSlot s    -> "Get current slot:" <+> pretty s
+
+instance ToObject NodeFollowerLogMsg where
+    toObject _ = \case
+        NewFollowerId fId  -> mkObjectStr "new follower id" (Tagged @"id" fId)
+        GetBlocksFor bId   -> mkObjectStr "Get blocks for " (Tagged @"id" bId)
+        LastBlock bId      -> mkObjectStr "Last block" (Tagged @"id" bId)
+        NewLastBlock bId   -> mkObjectStr "New last block" (Tagged @"id" bId)
+        GetCurrentSlot bId -> mkObjectStr "Get current slot" (Tagged @"id" bId)
+
+newtype FollowerID = FollowerID Int
+    deriving stock (Show, Eq, Ord, Generic)
+    deriving newtype (ToJSON, FromJSON, ToHttpApiData, FromHttpApiData, Integral, Enum, Real, Num, Pretty)
+
+data BlockEvent = NewSlot
+    | NewTransaction Tx
+    deriving (Generic, Show, ToJSON, FromJSON)
+
+instance Pretty BlockEvent where
+    pretty = \case
+        NewSlot          -> "Adding a new slot"
+        NewTransaction t -> "Adding a transaction " <+> pretty (Ledger.txId t)
+
+
+-- State --------------------------------------------------------------------------------------------------------------
+
+
+newtype NodeFollowerState = NodeFollowerState { _unNodeFollowerState :: Map FollowerID Int }
```

Comment:
```suggestion
newtype NodeFollowerState = NodeFollowerState { unNodeFollowerState :: Map FollowerID Int }
```

generally we use `un` for newtypes unless something like `run` makes more sense.

Reply:
I didn't write that code i only moved it around in the file :sweat_smile:  But makes sense to me :+1: